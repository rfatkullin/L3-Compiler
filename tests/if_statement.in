func main (args->[[char]]) -> int

	if (0 == 0) then
		print "cond1";
	endif;

	if (0 == 1) then
		print "cond2";
	else
		print "cond2_else"
	endif;

	if (0 == 1) then
		print "cond3";
	elseif (1 == 1) then
		print "cond3_elseif"
	endif;

	if (0 == 1) then
		print "cond4";
	elseif (1 == 2) then
		print "cond4_elseif";
	elseif (1 == 3) then
		print "cond4_elseif";
	else
		print "cond4_else";
	endif;

	main = 0;
endfunc
