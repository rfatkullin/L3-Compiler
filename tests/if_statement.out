cond1
cond2_else
cond3_elseif
cond4_else
